model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -188,7 +188,11 @@ def extract_logprobs(logprobs: Dict[int, Logprob]) -> Dict[int, float]:
     args = parser.parse_args()
 
     engine_args = AsyncEngineArgs.from_cli_args(args)
-    engine = AsyncLLMEngine.from_engine_args(engine_args)
+    try:
+        engine = AsyncLLMEngine.from_engine_args(engine_args)
+    except AsyncEngineDeadError as e:
```

Comment:
is this where it's crashed? i thought it crashes during serving requests

Reply:
+1, I think you can check on the engine's health with this fn - https://github.com/vllm-project/vllm/blob/main/vllm/engine/async_llm_engine.py#L714